sdk/go/client_test.go

```diff
@@ -101,3 +104,53 @@ func TestContainer(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "3.16.2\n", contents)
 }
+
+type muWriter struct {
+	mu sync.RWMutex
+	b  bytes.Buffer
+}
+
+func (w *muWriter) Write(b []byte) (int, error) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	return w.b.Write(b)
+}
+
+func (w *muWriter) String() string {
+	w.mu.RLock()
+	defer w.mu.RUnlock()
+	return w.b.String()
+}
+
+func TestConnectOption(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	w := muWriter{}
+	c, err := Connect(ctx, WithLogOutput(&w))
+	require.NoError(t, err)
+	defer c.Close()
+
+	_, err = c.
+		Core().
+		Container().
+		From("alpine:3.16.1").
+		FS().
+		File("/etc/alpine-release").
+		Contents(ctx)
+	require.NoError(t, err)
+	time.Sleep(1 * time.Second)
```

Comment:
Instead of sleeping you should be able to call `c.Close()` here. It'll wait for the engine to stop, which also waits for progress to finish writing.

With that change you should also be able to use a regular `bytes.Buffer` instead of `muWriter`. :)

Reply:
Those are some good suggestions.